Test failure with Python 3.14

### Environment

- **Qiskit version**: 2.2.1
- **Python version**: 3.14.0b2
- **Operating system**: macOS


### What is happening?

With Python 3.14 (actually 3.14.0b2), I see a test failure:

```
test.python.visualization.timeline.test_generators.TestGates.test_gen_sched_gate_with_finite_duration
-----------------------------------------------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "/Users/jake/code/qiskit/terra/tmp/test/python/visualization/timeline/test_generators.py", line 67, in test_gen_sched_gate_with_finite_duration
    self.assertDictEqual(ref_meta, drawing_obj.meta)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^

      File "/Users/jake/.local/share/uv/python/cpython-3.14.0b2-macos-x86_64-none/lib/python3.14/unittest/case.py", line 1224, in assertDictEqual
    self.fail(self._formatMessage(msg, standardMsg))
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

      File "/Users/jake/.local/share/uv/python/cpython-3.14.0b2-macos-x86_64-none/lib/python3.14/unittest/case.py", line 750, in fail
    raise self.failureException(msg)

    AssertionError: {'nam[110 chars]: '[[1.+0.j 0.-0.j]\n [0.+0.j 1.+0.j]]', 'para[14 chars], 0'} != {'nam[110 chars]: '[[ 1.+0.j -0.-0.j]\n [ 0.+0.j  1.+0.j]]', '[18 chars], 0'}
  {'bits': [<Qubit register=(1, "foo"), index=0>],
   'duration': 20,
   'label': 'n/a',
   'name': 'u3',
   'parameters': '0, 0, 0',
   't0': 100,
-  'unitary': '[[1.+0.j 0.-0.j]\n [0.+0.j 1.+0.j]]'}
+  'unitary': '[[ 1.+0.j -0.-0.j]\n [ 0.+0.j  1.+0.j]]'}
?                +       +           +      +
```

### How can we reproduce the issue?

Run the test suite with Python 3.14.

### What should happen?

The test should pass.  It looks like we're comparing two arrays by a string comparison instead of something like `np.allclose`, and it's a signed zero tripping us up.  Looks like the test is at fault, rather than Qiskit.

As part of this issue, we should flip the maximum-supported Python version in CI to be 3.14, up from 3.13.

### Any suggestions?

_No response_

--- a/.github/workflows/branch-protection.yml
+++ b/.github/workflows/branch-protection.yml
@@ -28,7 +28,7 @@ jobs:
       - run: ':'
     outputs:
       python-old: '3.10'
-      python-new: '3.13'
+      python-new: '3.14'
       runner-linux-x86_64: 'ubuntu-latest'
       runner-linux-arm64: 'ubuntu-24.04-arm'
       runner-macos-arm64: 'macos-15'

--- a/.github/workflows/docs_deploy.yml
+++ b/.github/workflows/docs_deploy.yml
@@ -23,7 +23,7 @@ jobs:
       - uses: actions/setup-python@v6
         name: Install Python
         with:
-          python-version: "3.13"
+          python-version: "3.14"
 
       - name: Install dependencies
         run: tools/install_ubuntu_docs_dependencies.sh

--- a/.github/workflows/on-nightly.yml
+++ b/.github/workflows/on-nightly.yml
@@ -19,7 +19,7 @@ jobs:
     uses: ./.github/workflows/test-linux.yml
     strategy:
       matrix:
-        python-version: ["3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14"]
         runner: ["ubuntu-latest", "ubuntu-24.04-arm"]
     with:
       python-version: ${{ matrix.python-version }}
@@ -30,7 +30,7 @@ jobs:
     uses: ./.github/workflows/test-mac.yml
     strategy:
       matrix:
-        python-version: ["3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14"]
         runner: ["macos-14"]
 
     with:
@@ -42,7 +42,7 @@ jobs:
     uses: ./.github/workflows/test-windows.yml
     strategy:
       matrix:
-        python-version: ["3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14"]
         runner: ["windows-latest"]
     with:
       python-version: ${{ matrix.python-version }}

--- a/.github/workflows/wheels-build.yml
+++ b/.github/workflows/wheels-build.yml
@@ -60,7 +60,7 @@ on:
       python-version:
         description: "The Python version to use to host the build runner."
         type: string
-        default: "3.13"
+        default: "3.14"
         required: false
 
       pgo:

--- a/asv.conf.json
+++ b/asv.conf.json
@@ -17,7 +17,7 @@
     "dvcs": "git",
     "environment_type": "virtualenv",
     "show_commit_url": "https://github.com/Qiskit/qiskit/commit/",
-    "pythons": ["3.10", "3.11", "3.12", "3.13"],
+    "pythons": ["3.10", "3.11", "3.12", "3.13", "3.14"],
     "benchmark_dir": "test/benchmarks",
     "env_dir": ".asv/env",
     "results_dir": ".asv/results"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,6 +40,7 @@ classifiers = [
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
     "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
     "Topic :: Scientific/Engineering",
 ]
 # These are configured in the `tool.setuptools.dynamic` table.
@@ -279,7 +280,7 @@ include = ["qiskit", "qiskit.*"]
 
 [tool.black]
 line-length = 100
-target-version = ['py310', 'py311']
+target-version = ['py310', 'py311', 'py312', 'py313']
 
 [tool.cibuildwheel]
 manylinux-x86_64-image = "manylinux_2_28"

